projects/ngx-charts-on-fhir/src/lib/fhir-mappers/simple-medication-mapper.service.ts

```diff
@@ -4,72 +4,53 @@ import { MedicationOrder } from 'fhir/r2';
 export type SimpleMedication = {
   medicationCodeableConcept: {
     text: string;
   };
   dateWritten: string;
 } & MedicationOrder;
 export function isMedication(resource: MedicationOrder): resource is SimpleMedication {
-  return !!(
-    resource.resourceType === 'MedicationOrder' &&
-    resource.prescriber?.reference &&
-    resource.dosageInstruction &&
-    resource.medicationCodeableConcept
-  );
+  return !!(resource.resourceType === 'MedicationOrder' && resource.dateWritten && resource.medicationCodeableConcept?.text);
 }
 
-/** Maps a FHIR Observation resource that has a single valueQuantity */
+/** Maps a FHIR MedicationOrder resource that only has a dateWritten and no supply duration */
 @Injectable({
   providedIn: forwardRef(() => FhirMappersModule),
 })
 export class SimpleMedicationMapper implements Mapper<SimpleMedication> {
   constructor(
     @Inject(TIME_SCALE_OPTIONS) private timeScaleOptions: ScaleOptions<'time'>,
-    @Inject(LINEAR_SCALE_OPTIONS) private linearScaleOptions: ScaleOptions<'linear'>,
-    @Inject(ANNOTATION_OPTIONS) private annotationOptions: ChartAnnotation
+    @Inject(CATEGORY_SCALE_OPTIONS) private categoryScaleOptions: ScaleOptions<'category'>
   ) {}
   canMap = isMedication;
-//   created Medication Data Layer 
-  map(resource: MedicationOrder): any {
-    let endDate = new Date(resource.dateWritten ? resource.dateWritten : "" )
-    endDate.setDate(endDate.getDate() + 2) 
-    const output =  {
+  map(resource: SimpleMedication): DataLayer {
+    return {
       name: resource?.medicationCodeableConcept?.text,
       category: 'Medication',
       datasets: [
         {
           label: resource?.medicationCodeableConcept?.text,
-        //   yAxisID: resource.valueQuantity.unit,
+          yAxisID: 'medications',
+          indexAxis: 'y',
           data: [
-            [
-              new Date(resource.dateWritten ? resource.dateWritten : "" ),
-              endDate
-            ],
+            {
+              x: new Date(resource.dateWritten).getTime(),
+              y: resource?.medicationCodeableConcept?.text,
+            },
           ],
         },
       ],
-    //   scales: {
-    //     timeline: this.timeScaleOptions,
-    //     [resource.valueQuantity.unit]: merge({}, this.linearScaleOptions, {
-    //       title: { text: resource.valueQuantity.unit },
-    //     }),
-    //   },
-    //   annotations: resource.referenceRange?.map<ChartAnnotation>((range) =>
-    //     merge({}, this.annotationOptions, {
-    //       display: true,
-    //       label: { content: `${resource.code.text} Reference Range` },
-    //       yScaleID: resource.valueQuantity.unit,
-    //       yMax: range?.high?.value,
-    //       yMin: range?.low?.value,
-    //     })
-    //   ),
+      scales: {
+        timeline: this.timeScaleOptions,
+        medications: merge({}, this.categoryScaleOptions, {
+          title: { text: 'Medications' },
+        }),
+      },
```

Comment:
Here is the scale configuration for category axis. it uses a hardcoded key (`medications`) so all medications will share the same scale. Each distinct y-value becomes the label for a tick on the category scale.

All of the examples in chart.js documentation used a `labels` array for category axis, but I found that is not actually necessary if we just strings as y-values. I think this way is much cleaner than using a separate labels array.